Xamarin.Forms.Core/Device.cs

```diff
@@ -103,11 +103,45 @@ public static void BeginInvokeOnMainThread(Action action)
 			PlatformServices.BeginInvokeOnMainThread(action);
 		}
 
-		public static double GetNamedSize(NamedSize size, Element targetElement)
+		internal static IDispatcher GetDispatcher(Element element)
 		{
-			return GetNamedSize(size, targetElement.GetType());
+			Page page = GetElementPage(element);
+			if (page != null)
+			{
+				return page.Dispatcher;
+			}
+			else
+			{
+				return default(IDispatcher);
+			}
 		}
 
+		internal static IDispatcher GetDispatcher()
+		{
+			return PlatformServices.GetDispatcher();
+		}
+
+		static Page GetElementPage(Element element)
+		{
+			if (element == null)
+			{
+				return null;
+			}
+			else if (element is Page page)
+			{
+				return page;
+			}
+			else
+			{
+				return GetElementPage(element.Parent);
+			}
+		}
+
+		public static double GetNamedSize(NamedSize size, Element targetElement)
+        {
+            return GetNamedSize(size, targetElement.GetType());
+        }
```

Comment:
It looks like it uses spaces instead of tabs

Reply:
Changed!